src/Cake.Core/ICakeEnvironment.cs

```diff
@@ -20,6 +18,12 @@ public interface ICakeEnvironment
         /// <value>The working directory.</value>
         DirectoryPath WorkingDirectory { get; set; }
 
+        /// <summary>
+        /// Gets the launch info.
+        /// </summary>
+        /// <value>The launch info.</value>
+        ILaunchInfo LaunchInfo { get; }
```

Comment:
```suggestion
        DirectoryPath LaunchDirectory { get; }
```

The `ILaunchInfo` is just an implementation detail to give you a way for you to store the `LaunchDirectory` early in the process, and inject into the places that need it. From the point of view of the API, for build scripts to access it, let's expose it in the same way that we do for `WorkingDirectory` above.